airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -644,17 +642,15 @@ def read_records(
         stream_slice: Mapping[str, Any] = None,
         stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
-        repository = stream_slice["repository"]
-        start_point_map = {
-            branch: self.get_starting_point(stream_state=stream_state, repository=repository, branch=branch)
-            for branch in self.branches_to_pull.get(repository, [])
-        }
+        start_point = self.get_starting_point(
+            stream_state=stream_state, repository=stream_slice["repository"], branch=stream_slice["branch"]
+        )
         for record in super(SemiIncrementalGithubStream, self).read_records(
             sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state
         ):
-            if record.get(self.cursor_field) > start_point_map[stream_slice["branch"]]:
+            if record[self.cursor_field] > start_point:
```

Comment:
This is potential bug for incremental refresh, the `record[self.cursor_field]` could be missing, can we try to use the `.get()` with the default value otherwise?

Reply:
as for me `.get()` is not silver bullet
as I see current logic assume that `record[self.cursor_field]` always exists.

I just stated this with more explicit `record[]`


previous implementation with .get()
will produce as for me less meanunfull error message:

```
TypeError: '<' not supported between instances of 'NoneType' and
```